### PR TITLE
Prototype transforms cleanup

### PR DESCRIPTION
**Pull request by pmeier**
Addresses minor comments from #5500.

### CODE REVIEW

File: torchvision/prototype/transforms/_augment.py [hunk 1]
```diff
@@ -99,8 +96,11 @@ def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
             return input
 
     def forward(self, *inputs: Any) -> Any:
-        if torch.rand(1) >= self.p:
-            return inputs if len(inputs) > 1 else inputs[0]
+        sample = inputs if len(inputs) > 1 else inputs[0]
+        if has_any(sample, features.BoundingBox, features.SegmentationMask):
+            raise TypeError(f"BoundingBox'es and SegmentationMask's are not supported by {type(self).__name__}()")
+        elif torch.rand(1) >= self.p:
+            return sample
```
Comment (datumbox) [inline torchvision/prototype/transforms/_augment.py:103]: Nit: (Can't comment on the right spot.) 2 lines below you pass `super().forward(*inputs)` but in other methods you do `return super().forward(sample)`. Choose one idiom.

File: torchvision/prototype/transforms/_augment.py [hunk 2]
```diff
@@ -99,8 +96,11 @@ def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
             return input
 
     def forward(self, *inputs: Any) -> Any:
-        if torch.rand(1) >= self.p:
-            return inputs if len(inputs) > 1 else inputs[0]
+        sample = inputs if len(inputs) > 1 else inputs[0]
+        if has_any(sample, features.BoundingBox, features.SegmentationMask):
+            raise TypeError(f"BoundingBox'es and SegmentationMask's are not supported by {type(self).__name__}()")
+        elif torch.rand(1) >= self.p:
```
Comment (datumbox) [inline torchvision/prototype/transforms/_augment.py:102]: Nit: Sometimes you do if/elif, other times below if/if. Choose one idiom for consistency.

File: torchvision/prototype/transforms/_auto_augment.py [hunk 3]
```diff
@@ -52,9 +55,10 @@ def fn(
             if type(input) in {torch.Tensor, features.Image} or isinstance(input, PIL.Image.Image):
                 return id, input
 
-            self._check_unsupported(input)
             return None
 
+        self._check_unsupported(sample)
```
Comment (datumbox) [inline torchvision/prototype/transforms/_auto_augment.py:60]: I think I should have been clearer here at https://github.com/pytorch/vision/pull/5500#discussion_r816514544. My fault.

Can we remove `_check_unsupported`, move the if statement in the `_extract_image` and parameterize `_extract_image` instead?

File: torchvision/prototype/transforms/_auto_augment.py [hunk 4]
```diff
@@ -480,9 +503,10 @@ def __init__(
         chain_depth: int = -1,
         alpha: float = 1.0,
         all_ops: bool = True,
-        **kwargs: Any,
+        interpolation: InterpolationMode = InterpolationMode.NEAREST,
```
Comment (datumbox) [inline torchvision/prototype/transforms/_auto_augment.py:506]: Augmix uses bilinear:
```suggestion
        interpolation: InterpolationMode = InterpolationMode.BILINEAR,
```

File: torchvision/prototype/transforms/_utils.py [hunk 5]
```diff
@@ -34,3 +34,15 @@ def get_image_dimensions(image: Union[PIL.Image.Image, torch.Tensor, features.Im
...
+def _extract_types(sample: Any) -> Iterator[Type]:
+    return query_recursively(lambda id, input: type(input), sample)
+
+
+def has_any(sample: Any, *types: Type) -> bool:
+    return any(issubclass(type, types) for type in _extract_types(sample))
+
+
+def has_all(sample: Any, *types: Type) -> bool:
+    return not bool(set(types) - set(_extract_types(sample)))
```
Comment (vfdev-5) [inline torchvision/prototype/transforms/_utils.py:48]: @pmeier maybe we could code `has_all` and `has_any` using torch tree_flatten and simplify the code:
```python
from typing import Any, Type

from torch.utils._pytree import tree_flatten


def has_any(sample: Any, *types: Type) -> bool:
    flat_sample, _ = tree_flatten(sample)
    return any(issubclass(type(obj), types) for obj in flat_sample)


def has_all(sample: Any, *types: Type) -> bool:
    flat_sample, _ = tree_flatten(sample)
    return not bool(set(types) - set([type(obj) for obj in flat_sample]))
```

**Pull request merged**